services/workflows-service/src/workflow/hook-callback-handler.service.ts

```diff
@@ -228,7 +226,7 @@ export class HookCallbackHandlerService {
     const customer = await this.customerService.getByProjectId(currentProjectId);
 
     if (comparedToReportId) {
-      const comparedToReport = await this.businessReportService.findFirst(
+      const comparedToReport = await this.businessReportService.findFirstOrThrow(
```

Comment:
the next line is not relevant anymore, you throw above if the report is not exists